tests/tests/test-subscription.ts

```diff
@@ -493,6 +494,29 @@ describeWithMoonbeam(
 
       expect(data).to.not.be.null;
     });
+
+    step("should not receive log when contract fails", async function () {
+      const subscription = web3Subscribe("logs", {});
+
+      await new Promise((resolve) => {
+        subscription.once("connected", resolve);
+      });
+
+      await sendTransaction(context, {
+        gas: "0xD99BF", // lower than what is needed by 1
+      });
+
+      const data = await new Promise((resolve) => {
+        createAndFinalizeBlock(context.polkadotApi);
+        let result = null;
+        subscription.once("data", (d) => (result = d));
+        setTimeout(() => resolve(result), 1000);
+        // wait for 1 second to make sure the notification has time to arrive.
+        // (it is not supposed to)
```

Comment:
```suggestion
        // wait for 1 second to make sure a notification would have time to arrive.
        // (This one is not supposed to arrive because the transaction ran out of gas.)
```